test/e2e/init_scale_zero_test.go

```diff
@@ -0,0 +1,139 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/test/logstream"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	"knative.dev/serving/pkg/apis/serving"
+	v1a1testing "knative.dev/serving/pkg/testing/v1alpha1"
+	"knative.dev/serving/test"
+	v1a1test "knative.dev/serving/test/v1alpha1"
+)
+
+func TestInitScaleZeroClusterLevel(t *testing.T) {
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := Setup(t)
+	tests := []struct {
+		name string
+		// configAutoscalerFlag indicates if the scale to zero on deploy flag is set to true
+		// or false in the config-autoscaler config map.
+		configAutoscalerFlag bool
+	}{{
+		name:                 "config-autoscaler true",
+		configAutoscalerFlag: true,
+	}, {
+		name:                 "config-autoscaler false",
+		configAutoscalerFlag: false,
+	}}
+	for _, tc := range tests {
+		func() {
+			t.Logf("Setting flag to %v on config-autoscaler ConfigMap.", tc.configAutoscalerFlag)
+			if tc.configAutoscalerFlag {
+				updatedConfigAutoscalerCM := setScaleToZeroOnDeployOnCluster(t, clients, "config-autoscaler")
+				defer restoreCM(t, clients, updatedConfigAutoscalerCM)
```

Comment:
We run our e2e tests in parallel and editing configmaps has global implications, which will lead to flakiness across the whole suite.

This should be in its own package, and we should set up the environment with the configmap setting enabled before running the test code.

Reply:
@mattmoor my understanding is that if `t.Parallel()` is not called, the test won't be run in parallel. The integration tests would have failed if this test is run in parallel with other tests in the suite, right?

Reply:
Honestly, the semantics of `t.Parallel()` are incredible bizarre to me.  The only reliable way I've found to avoid having things run in parallel is to split packages, which is why the scale tests live elsewhere.

I think the test system should own configuring the System Under Test (SUT) and the test should focus on verifying the desired behavior.  A good example of this is what we do for the auto-TLS e2e tests.  We effectively have one test (basically "Will it TLS?"), and we can run this in each of the three "modes" we can provision certs in today:
1. namespace wildcard cert (DNS01)
2. per-Route cert (DNS01)
3. per-Route cert (HTTP01)

The SUT changes, but the test does not.